app/domain/services/upload_student_exercises/service.rb

```diff
@@ -95,14 +95,15 @@ def process
           :course_excluded_exercise_group_uuids
         ).index_by(&:first)
 
-        # Get assignments that should have hidden feedback for each student
-        no_feedback_yet_assignments = Assignment
-                                        .where(student_uuid: student_uuids)
-                                        .where(aa[:feedback_at].gt(start_time))
-                                        .pluck(:student_uuid, :assigned_exercise_uuids)
+        # Get assignments are not yet due or do not yet have feedback for each student
+        anti_cheating_assignments = Assignment
+                                      .where(aa[:due_at].gt(start_time))
+                                      .or(Assignment.where(aa[:feedback_at].gt(start_time)))
+                                      .where(student_uuid: student_uuids)
+                                      .pluck(:student_uuid, :assigned_exercise_uuids)
 
```

Comment:
just double checking, but we do have indexes on `due_at` and `feedback_at` right?

Reply:
Yeah we have some indexes for them.